src/views/docs-view/server.ts

```diff
@@ -279,8 +279,14 @@ export function getStaticGenerationFunctions<
 				menuItems: navDataWithFullPaths as EnrichedNavItem[],
 				title: currentRootDocsPath.shortName || currentRootDocsPath.name,
 			}
-			// If the title is not hidden for this rootDocsPath, include it
-			if (currentRootDocsPath.visuallyHideSidebarTitle) {
+			/**
+			 * In some cases, the first nav item is a heading.
+			 * In these case, we'll visually hide the sidebar title,
+			 * since it will redundant right next to the authored title.
+			 */
+			const firstItemIsHeading =
+				typeof navDataWithFullPaths[0]?.heading == 'string'
+			if (firstItemIsHeading) {
```

Comment:
Whether the first nav node is a heading felt like the condition under which we want to visually hide the sidebar's title. At least, this lines up for all the existing use cases in Terraform docs 👍